src/dotnet-interactive-vscode-insiders/src/notebookControllers.ts

```diff
@@ -170,16 +173,16 @@ export class DotNetNotebookKernel {
 
                 return client.execute(source, getSimpleLanguage(cell.document.languageId), outputObserver, diagnosticObserver, { id: cell.document.uri.toString() }).then(async () => {
                     await outputUpdatePromise;
-                    endExecution(cell, true);
+                    endExecution(client, cell, true);
```

Comment:
do we need to add the execution count management to the client? Is looks something the controller should manage since is a mapping with the notebook id and also the controller manages the restart of the kernel associated with a client

Reply:
We need to have it on the kernel because:

1. We need to be able to reset to `1` on kernel restart and at that point we only have access to the client.
2. We need to keep that data in the client's metadata so if the user does a save-as .ipynb (which goes through our own code) we can populate the `execution_count` field.

Reply:
ouch .. I see, approving then ..